unity/Assets/Scripts/BaseFPSAgentController.cs

```diff
@@ -2972,5 +2981,36 @@ void OnDrawGizmos()
             // }
         }
         #endif
+
+        public void TestActionDispatchNoopServerAction(ServerAction action) {
+            actionFinished(true, "serveraction");
+        }
+
+        public void TestActionDispatchNoopAllDefault(float param12, float param10=0.0f, float param11=1.0f) {
+            actionFinished(true, "somedefault");
+        }
+
+        public void TestActionDispatchNoopAllDefault(float param10=0.0f, float param11=1.0f) {
+            actionFinished(true, "alldefault");
+        }
+
+        public void TestActionDispatchNoop(bool param3,  string param4="foo") {
+            actionFinished(true, "param3 param4/default " + param4);
+        }
+
+        public void TestActionDispatchNoop(string param6, string param7) {
+            actionFinished(true, "param6 param7");
+        }
+
+        public void TestActionDispatchNoop(bool param1, bool param2) {
+            actionFinished(true, "param1 param2");
+        }
+
+        public void TestActionDispatchNoop(bool param1) {
+            actionFinished(true, "param1");
+        }
+        public void TestActionDispatchNoop() {
+            actionFinished(true, "emptyargs");
+        }
```

Comment:
Maybe we should consider also having a test agent? just this class is becoming really big

Reply:
I agree.  I am going to create a separate issue that addresses Agent refactoring with the goal of making all the agentType, subclassing a little better.  I'm not sure what approach we should take, but I think we need to make things a little more extensible.